Content Libraries: documentation

catong opened the pull request:
This is the initial draft of documentation for the Content Library feature in the "Building and Running a Course" guide. 
@antoviaque @marcotuts  FYI @smagoun @mhoeber @srpearce @lamagnifica 
Please flag anyone I've missed who might want to review.
Please review by EOD Monday Jan 12 (sorry for the short notice)!

For easier reading, here is an HTML version: 
http://draft-content-libraries.readthedocs.org/en/latest/developing_course/libraries.html

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Content Libraries 
+##################
+
+In Studio, use content libraries to build a pool of components that can be
```
In Studio, you(?) In Studio, course teams(?)

lamagnifica replied:
pull "that can be shared across multiple courses" out into a separate, later sentence? it seems to me that this feature will be used primarily to randomize content delivery within a course, and then--potentially but not necessarily--built up into a cooperative cross-course library. @explorerleslie , what's your take?

explorerleslie replied:
agree with @lamagnifica. We need to put the focus on the randomized problem banks part of this, less so on the content libraries piece. We had a lot of confusion at the last product council for the same reason -- when people think about "content libraries", they think it will solve all the pain around sharing course material across courses and people, but that's not the feature we're building right now.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+HTML components, problems, and video components to a library. Peer assessment
+and discussion components are not supported in randomized content blocks.
+
+Create randomized assessments in your course by adding a randomized content
```
You create...

antoviaque commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+If you modify the contents of a library that is references by randomized content
+blocks in one or more courses, you can bring the course components up to date
+with the version in the library.
```
It could be important to warn that this should be avoided when the randomized assessment has already been published - just like any type of problem, once students have started answering or interacting with it, changing the problem definition can be problematic.

catong replied:
Good point, thanks!

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+#. Log in to Studio.
+#. Click **New Library**.
+#. Enter mandatory information for your new library, then click **Create**.
```
NIT - Enter in / Provide  (?)

lamagnifica replied:
mandatory --> required

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Edit Components in a Library
+****************************
+
+You can perform the same tasks with components in a library as you can with
```
NIT - similar tasks (you can't drag drop, but this isn't necessarily critical enough to note here.)

antoviaque commented:
:+1: besides the comment I left.

@bradenmacdonald Could you also review this today?

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+View the Matching Contents of a Randomized Content Block
+*********************************************************
+
+In the course outline, in the unit that uses a randomized content block, you can
```
From a unit page that uses a randomized content block you can see which library components were matched by clicking on the view button for the randomized content block.  (At a minimum I think we could drop the course outline reference from this.)

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Getting the Latest Version of Library Content
+*********************************************
+
+If you modify the contents of a library that is references by randomized content
```
s/references/referenced

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  The randomized content block is brought up to date with the latest contents of
+  the library or libraries that it references.
+
+* If you edited the components in the course outline so that they are different
```
If you edited the components in the *randomized content block view* (?) you will lose changes which were made to the editor and content of the block, but you will not lose the setting overrides you edited in the settings tab.

marcotuts commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Editing Components in Randomized Content Blocks
+******************************************************
+
+From within your course outline, in the unit that uses a randomized content
```
From within the randomized content block view, nested inside its corresponding unit page, 

(no specific wording suggested here, but this is a general question about how we refer to the nested view for this RCB component. )

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+In Studio, use content libraries to build a pool of components that can be
+shared across multiple courses to create randomized assessments. You can add
+HTML components, problems, and video components to a library. Peer assessment
+and discussion components are not supported in randomized content blocks.
```
just confirming -- can't add other  advanced components?

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+and discussion components are not supported in randomized content blocks.
+
+Create randomized assessments in your course by adding a randomized content
+block to a unit, and specifying the number and type of problems that each
```
This is somewhat confusing right now. Is a "randomized content block" the same as the "specified content library"? or is there another component of this procedure, where the libraries get associated with the content block? Either way, I recommend breaking up the sentence:
You create randomized assessments in your course by adding a randomized content block to a unit. You then specify the number and type of problems to assign to each student. The problems are drawn from the content library or libraries that you specify for the content block. (?)

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  .. image:: ../Images/ContentLibraryID.png
+     :alt: The Library ID for the new library is shown the sidebar
+
+.. note:: You might want to copy or make a note of the Library ID for the    new
```
spacing at end of line

mhoeber replied:
I'd delete "You might want to . . ."

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  * For **Library Code**, enter an identifier for your library that is unique
+    within your organization. This code becomes part of the URL for your
+    library, so do not include spaces or special characters in the code.
+
```
Assuming none of these would ever be **student** visible.  But if they are we'd want to add a caution about that.

bradenmacdonald replied:
Correct. Neither Library code nor Library ID or anything to do with libraries are ever seen by students.

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+You cannot delete a library after creating it. Instead, you can simply
+discontinue use of an unwanted library. To do so, first make sure that none of
+its components are in use in any courses, then delete all components in the
```
how would you make sure none of it's components are used in courses? Do you have to go through each course? Or is it reported in library (which would be nice)?

bradenmacdonald replied:
In the future it will be reported in the library. Unfortunately that hasn't been implemented in our MVP.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+********************
+Create a New Library
+********************
+
```
I think you do need to preface this with a statement that libraries are not associated with a single course.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+To create a new :ref:`library<Content Libraries>`, follow these steps.
+
+#. Log in to Studio.
+#. Click **New Library**.
```
If you are already logged in to Studio, how do you get to the page where you manage libraries? who else can access the libraries that you create?

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+.. note:: Enter new library information carefully. After you create a new
+     library, you cannot delete it. The Library Code becomes part of the URL
+     for your library.   
+
```
is there a length limit, as there is for course name/run/etc. info for courses?

bradenmacdonald replied:
Yes, 65, same as for courses (TNL-889) / https://github.com/edx/edx-platform/pull/6046/files#diff-8ad11258fb6ce404695c878f864ceef2R11

lamagnifica replied:
see language here @catong http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/building_course/creating_new_course.html#create-a-new-course

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+You see the new library, to which you can now add components. For details about
+adding components to a library, see :ref:`Add Components to a Library`.
+
+The system-generated Library ID for the new library is shown in the sidebar.
```
is shown --> appears

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+.. note:: You might want to copy or make a note of the Library ID for the    new
+   library. When you add a randomized content block to a course unit, you use
+   the Library ID to specify this library as a source for the randomized
+   content.
```
raises the question, what's the purpose of the required unique Library Code if what you need is the system-generated Library ID?

bradenmacdonald replied:
If i understand your question correctly: The library code becomes part of the library ID, and its purpose is to make the library ID more readable. The library code is not unique by itself (it is only unique when combined with the organization to form the library ID).

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Edit a Library
+**************
+
+After you create a new library, the only change you can make to the initial
```
strike new

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+After you create a new library, the only change you can make to the initial
+library information is to the name. However, at any time, you can make changes
+to the components in your library, including adding or deleting components, and
```
components, and --> components or

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   
+#. Click the **Edit** icon next to the library name. 
+   
+   The library name field becomes editable.
```
I'd move this after the image

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. Click **Add Component**, then click the component type that you want to add
+   under **Add New Component**.
+
+For more information about the various types of components, see these topics.
```
strike various, maybe instead "For more information about the types of components you can add to a library,..."

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+After you add a component to a library, you can edit its settings, which are
+retained when the component is selected from the library and used in a course.
+Within a course, you can further edit the component settings or revert to the
+library settings. For details, refer to :ref:`Edit Components in a Library` and
```
This isn't clear to me. I think that this is saying that a component can have settings that are specific to the library, and potentially different from what they are in the course. I can't figure out how that would work, though. Would you somehow be able to add the identical component explicitly to the course? would that meant that all students would see the content with its course settings, and that a subset of students would see the same content with its library settings?

bradenmacdonald replied:
@lamagnifica  You can change things like the "display_name" and "weight" when using it in a course.

So a library might have two problems: "Problem 1" (weight 1) and "Problem 2" (weight 1). These are the values set by the library author when the problems are edited within the library.

Teacher A might use the library problems for a midterm in her Automotive Mechanics course, but she edits them within the course so that they are called "Final Question" and "Final Question", and so that they each have a weight of 10. That way her students will see one question (with two random possibilities) called "Final Question" and it will have a weight of 10.

Meanwhile Teacher B might use the library problems for a homework assignment in her Bus Engines class. The edits that teacher A made will not affect teacher B's course. Teacher B's students will see "Problem 1" and "Problem 2" and the problems will have a weight of 1.

Does that help at all?

catong replied:
Thanks, @bradenmacdonald! An example would be helpful here, it seems. I'll work on one -- thanks for the headstart on that :-)

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. Click **Libraries**, then click the library whose components you want to
+   view.
+
+The components in the library are displayed in the order in which they were
```
are displayed --> display

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+The components in the library are displayed in the order in which they were
+added, with the most recently added at the bottom. Each page shows up to 10
+components. If your library has more than 10 components, additional components
+are shown on other pages.
```
I think you can strike one of these last 2 sentences.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+are shown on other pages.
+
+The range of the components shown on the current page, and the total number of
+components, are shown at the top of the page.
```
this seems like overkill, should be self-explanatory in the UI, yes?

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  .. image:: ../Images/file_pagination.png
+     :alt: Image showing a pair of page numbers with the first number circled
+
+To view the list of matching components in the library, see :ref:`View the
```
what is a matching component? why would I need to view the list of matching components in the library, and how is that different from viewing the entire contents?

bradenmacdonald replied:
It's possible to use a "filter" so that only some components are used. For example you can say "use components from library X but only multiple choice problems". In that case, the matching components (multiple choice) are a subset of the entire contents of the library.

catong replied:
I've reworded to clarify.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+* :ref:`Duplicate a Component`
+* :ref:`Delete a Component`
+
+.. note:: If you modify components in your library that are in use in a course,
```
yeah, this connection between what you've put into the library and what you might separately have put in your course is quite mysterious to me...

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+There are three levels of access for libraries:
+
+* **User** -- Users can view library content and can reference or use library
```
if I'm course staff who hasn't been granted this access, when I'm adding components, I don't have the option of adding components from the library? just confirming.

lamagnifica replied:
what is the difference between referencing and using a library?

bradenmacdonald replied:
If you're a course staff who has been granted this "User" access to a library then yes you can "add components from the library" to your course. However you may not add components *to* the library nor edit the components in the library.

It's literally like an actual library - if you're a library patron ("User") you can borrow books and photocopy them for use in your course, but you are not allowed to write in the books nor add new books to the shelves of the library.

There is no different between "referencing" and "using" a library AFAIK.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Delete a Library
+*****************
+
+You cannot delete a library after creating it. Instead, you can simply
```
You cannot delete a library after creating it.  --> You cannot delete a library.
Strike "simply" (editorializing)

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+
+=========================
+Add Staff or Admin Access
```
how do you create the first admin? if you need to be an admin to create an admin.

bradenmacdonald replied:
The first admin is generally assigned automatically and is the person who created the library. There are also users who have admin access to all courses and libraries on the system (confusingly called "Global Staff" but they have Admin powers) who can always edit the admins.

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  - For **Display Name**, enter name that you want students to see for this
+    block.
+
+  - For **Libraries**, enter the unique Library ID found in the upper right of
```
assuming you are entering directly, not picking from a list.  So may want to say before step 1 to get and copy the library ID.

lamagnifica replied:
+1

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+    .. image:: ../Images/ContentLibraries_SelectProblemType.png
+     :alt: The Edit icon to the right of the Library Name    
+
+  - For **Scored**, from the drop down list select **True** or **False** to
```
not sure I understand this -- wouldn't grading be set by containing subsection?  If subsection is not graded, can you still  set Scored to true?  If subsection is not graded, what does true mean?

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+* If you edited the components in the course outline so that they are different
+  from the original version in the library, you do not lose your changes if you
+  click **Update now**. In this case, the changes that were made in the randomized
```
when you do this, is unit set to draft?  Do you need to republish the unit? Or is this outside of the publish settings?

bradenmacdonald replied:
Yes, that should set the unit to draft and the unit will need to be republished.

mhoeber commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+******************************************************
+
+From within your course outline, in the unit that uses a randomized content
+block, you can edit component settings in the same way as you do for any other
```
"you can edit **randomized content block** settings in the same way . . ."

catong replied:
It's correct as written, but I'm currently rewriting to try to add clarity throughout. Here you are editing the settings for *components* provided within the randomized content block, NOT the settings of the block itself.

mhoeber commented:
@catong  this is great, thanks.  I approached this as an ignorant user (cause I am) and asked some questions, which may or may not be relevant.  Let me know if I can help with follow up on these.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Give Other Users Access to Your Library
+***************************************
+
+You can give other Studio users access to your library. Depending on their role,
```
role --> roles

bradenmacdonald replied:
Actually, sorry to muddy the waters but *both* courses and libraries have levels of access. Also the pre-existing roles that grant privileges to work with courses have been extended to apply to libraries as well.

Courses and libraries share the two basic levels of access ("Roles" internally). They can be assigned on a per-user and per-course (or per-library) basis and work the same way in both cases:
* Staff
* Admin

Libraries have an additional third level of access:
* User (read-only)

Someone can work only with libraries and never touch a course, yes. Also vice versa.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+You can give other Studio users access to your library. Depending on their role,
+additional library users can view and use library content in courses, edit
+library content, or add and manage other library users. All course team members
+must be registered with Studio and have an active account.
```
Not sure what the purpose of this last sentence is. Much earlier in the chapter, I'd suggest a straightforward, 
"To work with content libraries, you must register and activate a user account in Studio."
Here, possibly the sentence that is needed would state that to work with a library AND with a course, you have to have a privileged role in that course(?)
Question: can anyone create a library in Studio on both edge and edx.org? no restriction on edx.org to global staff privileges (edX employees only)?

bradenmacdonald replied:
Creating a library requires the same permissions as creating a course. I'm not sure how edx.org/edge are set up, but there is an option to restrict creation of courses (& libraries) to certain users only.

catong replied:
Thanks, good note about who can create a library.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+library content, or add and manage other library users. All course team members
+must be registered with Studio and have an active account.
+
+There are three levels of access for libraries:
```
I like calling these "levels of access" as distinct from the course-related "roles".

I always try to avoid enumerating how many items are in a list, as they have the darndest tendency to change over time! 
Maybe rewrite as a complete sentence, "These are the levels of access for libraries."

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  they can add and remove other team members. There must be at least one user
+  with Admin privileges in a library.
+
+.. note:: New library members are first added with User privileges only. After
```
rewrite in active voice

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+To grant initial User access to a team member, follow these steps.
+
+#. Ensure you have Admin access. 
```
...access to the library." (how do you do that?)

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+To grant initial User access to a team member, follow these steps.
+
+#. Ensure you have Admin access. 
+#. Ensure that the new team member has an active Studio account. 
```
how?

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   are adding this user.
+#. From the **Settings** menu select **User Access**.
+#. On the **User Access** page, click **Add a New User**.
+#. Enter the new user's email address, then click **ADD USER**.
```
what if you add someone who doesn't have a Studio account yet, or who hasn't activated? will the access level be waiting for them (like an instructor enrolling a student by email address on the Instructor Dashboard)?

bradenmacdonald replied:
No, the access won't be waiting for them. You can only grant them access if they're already registered (not sure about registered but not activated - that may work).

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. Enter the new user's email address, then click **ADD USER**.
+   
+   The new user is added to the list of people with access to the library, with
+   their role indicated.
```
singular/plural, passive

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+=========================
+
+After you add new members to the library and give them User access, you (or
+another library user with the Admin role) can grant them additional privileges.
```
privileges --> levels of access

catong replied:
Good point, thanks

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+another library user with the Admin role) can grant them additional privileges.
+To grant a user Admin privileges, you must first assign them to the Staff role,
+then assign them to the Admin role.
+
```
"The levels of access for libraries are hierarchical. After you give someone the User level of access, you can give her the Staff level of access. Only people who already have the Staff level of access can be given the Admin level of access."

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. On the **User Access** page, locate the user to whom you are giving
+   additional privileges. 
+
+  - If they are currently in the User role, click **Add Staff Access**. 
```
they are --> he is (both lines)

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+Remove Staff or Admin Access
+============================
+
+After you have granted a library team member Staff Access or Admin Access, you
```
lower case "Access"

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+============================
+
+After you have granted a library team member Staff Access or Admin Access, you
+(or another library user with the Admin role) can reduce their level of access.
```
their --> her

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+============================
+Remove Staff or Admin Access
+============================
```
if you give some User access, can you ever get rid of them? or does this truly only work for Staff and Admin peeps?

bradenmacdonald replied:
Yes there is a delete button (trash can icon) you can click to remove them from the library completely.

catong replied:
Added a "Remove User from the Library" procedure

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  - If they are currently in the User role, click **Add Staff Access**. 
+  - If they are currently in the Staff role, click **Add Admin Access**.
+
+  The user's display listing is updated to indicate their new role. In
```
suggest changes a la line 271

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. On the **User Access** page, locate the user whose access level you are
+   changing. If they are currently in the Staff role, click **Remove Staff
+   Access**. If they are currently in the Admin role, click **Remove Admin
+   Access**.
```
they are --> she is (both times)

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   Access**. If they are currently in the Admin role, click **Remove Admin
+   Access**.
+
+   The user's display listing is updated to indicate their new role. 
```
see line 271

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   The user's display listing is updated to indicate their new role. 
+
+.. note:: There must always be at least one Admin for a library. If there is
+   only one user with the Admin role, you cannot remove them from the Admin role
```
them --> him

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+After you create one or more content libraries and add components to them, you
+can create randomized assessments in your course that draw a specified number
+and type of problem randomly from your content libraries.
```
does this only work for problem components? what about HTML and video? or, is this section really only about randomized assessments (inferring that only problem components van be involved?)

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   
+.. note:: After adding a randomized content block to your course, be aware that
+   if components in the source library or libraries change, you must manually
+   update the components in the course outline if you want to keep the components
```
this is so confusing. How can a component be in both the library and in the course outline?

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. Select **Settings**, then **Advanced Settings**.
+
+#. In the **Advanced Module List** field, place your cursor between the
+   supplied pair of braces.
```
I think those are brackets (square [ ]) rather than braces (curly { })

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+#. Click **Save Changes**. 
+   
+   Studio reformats the name:value pair you just entered to indent it on a new
```
in this case, you just entered a value, not a name:value pair

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+
+  .. image:: ../Images/ContentLibraries_AdvancedSetting.png
+     :alt: Advanced Module policy key "library_content"
+
```
is this one of those settings that might have a whole list of other values? if there are any other values, you also need to add a comma so that the values in the brackets are arrayed as a comma-separated list.

catong replied:
No other settings

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. Click **Advanced**, then click **Randomized Content Block**.
+   The randomized content block is added to your unit.
+
+#. Click the **Edit** icon or click the **Select a Library** link.
```
I'd just present one of these two options (if they're equivalent).

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  - For **Count**, enter the number of problems to be drawn from the specified
+    content library or libraries for each student.
+
+  - For **Display Name**, enter name that you want students to see for this
```
...enter ^the name...

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   unless you first assign another user to the Admin role.
+
+
+.. _Use Components from Libraries in a Course:
```
Have you considered organizing the information under this heading in the same way as other advanced components in the "Creating Exercises and Tools" chapter, or maybe even moving this into a separate rst file in that part of the guide? http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/exercises_tools/index.html

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+.. _Add a Randomized Content Block to Your Course:
+
+=============================================
+Add a Randomized Content Block To Your Course   
```
lower case "to"

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+#. In the randomized content block settings, specify the details of the content
+   you want to add in this block.
+
+  - For **Count**, enter the number of problems to be drawn from the specified
```
suggest simplifying by replacing "be drawn from the specified content library or libraries for" with "present to"

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+    content library or libraries for each student.
+
+  - For **Display Name**, enter name that you want students to see for this
+    block.
```
I think this is the first time we are exposing "block" in the UI. Are blocks materially different from other types of advanced components? I'm wondering if we should always use "randomized content block" or if "block" will clearly be synonymous with randomized content blocks.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+    the Library page in Studio. To select problems from more than one content
+    library, click **Add** to enter each additional Library ID.
+
+  - For **Problem Type**, from the drop down list select a specific type of
```
I think Sylvia wants us to standardize on drop-down hyphenated.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+.. _View the Matching Contents of a Randomized Content Block:
+
+*********************************************************
+View the Matching Contents of a Randomized Content Block
```
is this really a way of filtering the contents of the referenced library/libraries to show only problems of the type you selected on line 423?  if so, this title isn't really conveying that to me since I don't know what is being matched...

marcotuts replied:
Once you configure a RCB, and there are matches found from the library, you can View those matches by clicking the View button shown on the Unit page containing that RCB. This nested view is similar to the view we render for A/B experiments. On this nested view, we have a statement at the top of the page that tells you that the visible components are all eligible for inclusion in the RCB. That message also tells you how many problems will be selected for each student, which is something you have by then configured in the RCB block itself.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+assign problems of the "Multiple Choice" type, you see all the multiple choice
+problems that exist in the referenced library or libraries.
+
+#. In Studio, navigate to the unit containing the randomized content block that
```
to Marco's point, you could move the Course Outline page instruction here. 
1. In Studio, open the Course Outline for your course.
2. Locate the unit that contains the randomized content block.
3. ...

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+****************************************
+
+To view components from a randomized content block as a student would see them,
+follow the steps described in the :ref:`Preview a Unit` topic.
```
you could be a tiny bit more specific here, noting that both the **Number** and **Problem Type** settings are used to prepare the Preview view, just as they are for a student.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+*********************************************
+Getting the Latest Version of Library Content
+*********************************************
+
```
maybe way earlier than this, state that: 
When you add a RCB to a course and associate a library with it, the version of the library that the course uses is static and does not automatically reflect any further changes made to the library. You must manually refresh the RCB to use the latest version of the library. This ensures that changes can be made to the library without regard to whether any of the courses that use that library are running. Changes to a RCB in a course that is running can affect student grades and your ability to analyze course data.

catong replied:
Helpful wording on impacts to updates on grades and course data. Thanks!

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+      Update Now link circled.
+
+  To update your randomized content block components to the latest versions in the content library, click **Update now**.
+
```
how do you leave it alone? this might be just as important for teams to know...

bradenmacdonald replied:
The only way at the moment is to avoid clicking on the "Update Now" link. There's no "Dismiss" option yet or anything.

catong replied:
:-) Yes, I had drafted a "don't update" the library version, but had second thoughts about it. I think it would be useful to add back some mention of having the option not to update to the library version.

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+  click **Update now**. In this case, the changes that were made in the randomized
+  content block in the course outline are kept, but a **Clear** button becomes
+  available next to the changed field in the component.
+
```
I am entirely confused

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+.. _Edit Components in Randomized Content Blocks:
+
+******************************************************
+Editing Components in Randomized Content Blocks
```
there are components inside the RCB? 
I am so confused!

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+   randomized content block.
+
+To view the list of matching components in the library, see :ref:`View the
+Matching Contents of a Randomized Content Block`.
```
I'd suggest using the same noun (component or content) but this is the part that is confusing me a lot

lamagnifica commented on en_us/course_authors/source/developing_course/libraries.rst:
```diff
@@ -0,0 +1,550 @@
+a **Clear** icon is shown next to the setting field.
+
+ .. image:: ../Images/ContentLibraries_ResetComponentField.png
+    :alt: Clear button in the course component field reverts value to library value.
```
button --> icon

lamagnifica commented on en_us/course_authors/source/developing_course/index.rst:
```diff
@@ -22,5 +24,6 @@ Index`.
    course_subsections
    course_units
    course_components
+   libraries
```
I'm not sure that this is the right place for this section. Content Libraries might actually be described in their own chapter, as they are created and maintained entirely independently of courses. Whether or not a course makes use of content libraries is entirely optional, but I don't think that's ever actually stated in this draft. Pretty much everything else in this chapter is required. Here's a proposed alternate organization for the content you have in this chapter, as a devil's advocate:

New Chapter: Setting Up a Content Library
 What they are, use cases
 Creating them
 Adding components to them
 Giving other people access to them
 Viewing
 Editing
 Deleting

I'd make this hierarchically equal to the Setting Up Your Course chapter, but put it after all info on courses (after the Running Your Course chapter). 

Existing chapter Creating Exercises and Tools: add a separate alphabetized Randomized Problem Bank rst
 What they are, how they relate to libraries with xref, how they appear to students, how they are graded
 Adding one to a course
 Viewing relevant content
 Refreshing content to get changes made to the library
 Customizing content on a per-course basis

catong replied:
Good points. Thanks!

lamagnifica commented on en_us/course_authors/source/developing_course/index.rst:
```diff
@@ -13,6 +13,8 @@ Course Content Index`.
 For information on creating exercises and tools, see :ref:`Exercises and Tools
 Index`.
 
+For information on creating content libraries from which you can randomly draw content for randomized assessments, see :ref:
```
complete

lamagnifica commented:
Great feature, but some areas are confusing right now. Many suggestions in this first draft, but I know the deadline is looming on this. Let me know if I can help in any way.

bradenmacdonald commented:
@catong Sorry for not mentioning this earlier: It looks like this doesn't yet explain that you can only use Randomized Content Blocks on courses that are stored in split mongo. It's entirely possible that people might try to use a Randomized Content block in an Old Mongo course and run into strange errors.

Unfortunately, it may be tricky to explain that in a non-technical way, as Studio doesn't really indicate which courses are split mongo or old mongo. AFAIK the only way is to look at the course URL/ID and see if it starts with "course-v1" or not.

catong commented:
@bradenmacdonald thanks for the update. I'll get with @marcotuts and figure out how to inform users about this.

catong commented:
@antoviaque @marcotuts @lamagnifica @mhoeber 
This 2nd draft includes these changes:
- incorporated review comments
- restructured the content to separate information for 2 audiences: 1) that the library creation piece is in "Creating Course Content", and 2) the use of randomized content blocks in course outline is covered under "Creating Exercises and Tools" > "Randomized Content Blocks".
- adds a note in the Content Libraries overview that content libraries are only supported for courses with the particular course ID format that indicates use of split mongo.

Updated HTML is here: http://draft-library-doc-test.readthedocs.org/en/latest/creating_content/libraries.html

catong commented:
Closing this PR to create a new one to merge to Saudi-specific branch.
